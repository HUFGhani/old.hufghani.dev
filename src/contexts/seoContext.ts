import { createContext, useContext } from 'react'

interface SEOContextType {
  data: {
    site: {
      siteMetadata: {
        author: string
        url: string
        twitter: string
        titleAlt: string
        siteUrl: string
        siteLanguage: string
        shortName: string
        pathPrefix: string
        description: string
      }
    }
  }
}

export const SEOContext = createContext<SEOContextType | undefined>(undefined)

export const useSEOContext = () => useContext<SEOContextType | undefined>(SEOContext)
