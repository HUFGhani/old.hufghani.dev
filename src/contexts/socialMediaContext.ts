import { createContext, useContext } from 'react'
import SocialMediaInterface from '../interface/socialMediaInterface'

export const SocialMediaContext = createContext<SocialMediaInterface | undefined>(undefined)

export const useSocialMediaContext = () => useContext<SocialMediaInterface | undefined>(SocialMediaContext)
