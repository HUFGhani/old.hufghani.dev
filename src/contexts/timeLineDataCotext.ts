import { createContext, useContext } from 'react'
import TimelineItemInterface from '../interface/timelineItemInterface'

export const TimeLineDataCotext = createContext<TimelineItemInterface | undefined>(undefined)

export const useTimeLineDataCotext = () => useContext<TimelineItemInterface | undefined>(TimeLineDataCotext)
