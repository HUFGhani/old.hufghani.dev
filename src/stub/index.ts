export const SiteMetaData = {
  author: `Hamza Umar Farooq Ghani`,
  url: `https://hufghani.dev`,
  twitter: `@the_ghani`,
  titleAlt: `Hamza Ghani - Personal Website`,
  siteUrl: `https://hufghani.dev`,
  siteLanguage: `en`,
  shortName: `HUFGHANI`,
  pathPrefix: `/`,
  description: `personal Website`,
}
