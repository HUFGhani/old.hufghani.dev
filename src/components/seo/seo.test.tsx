import { render, waitFor } from '@testing-library/react'
import React from 'react'
import SEO from '.'
import { SEOContext } from '../../contexts/seoContext'
import { SiteMetaData } from '../../stub'

const siteTitle = 'Hello World'

function renderSEO() {
  return render(
    <SEOContext.Provider value={SiteMetaData}>
      <SEO pageTitle={siteTitle} />
    </SEOContext.Provider>
  )
}

describe('<SEO/>', () => {
  it('should render siteTitle', async () => {
    renderSEO()
    await waitFor(() => {
      expect(document.querySelector('meta')).toBeInTheDocument()
    })
    expect(document.querySelectorAll('meta')).toMatchSnapshot()
  })
})
