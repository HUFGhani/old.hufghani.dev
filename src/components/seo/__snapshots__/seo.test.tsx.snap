// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SEO/> should render siteTitle 1`] = `
NodeList [
  <meta
    charset="utf-8"
    data-react-helmet="true"
  />,
  <meta
    content="personal Website"
    data-react-helmet="true"
    name="description"
  />,
  <meta
    content="HUFGHANI"
    data-react-helmet="true"
    name="apple-mobile-web-app-title"
  />,
  <meta
    content="HUFGHANI"
    data-react-helmet="true"
    name="application-name"
  />,
  <meta
    content="Hamza Umar Farooq Ghani"
    data-react-helmet="true"
    name="author"
  />,
  <meta
    content="https://hufghani.dev"
    data-react-helmet="true"
    property="og:url"
  />,
  <meta
    content="Hello World"
    data-react-helmet="true"
    property="og:title"
  />,
  <meta
    content="personal Website"
    data-react-helmet="true"
    property="og:description"
  />,
  <meta
    content="summary_large_image"
    data-react-helmet="true"
    name="twitter:card"
  />,
  <meta
    content="@the_ghani"
    data-react-helmet="true"
    name="twitter:creator"
  />,
  <meta
    content="Hello World"
    data-react-helmet="true"
    name="twitter:title"
  />,
  <meta
    content="personal Website"
    data-react-helmet="true"
    name="twitter:description"
  />,
  <meta
    content="width=device-width, initial-scale=1.0, viewport-fit=cover"
    data-react-helmet="true"
    name="viewport"
  />,
]
`;
